airflow/models/log.py

```diff
@@ -73,7 +73,7 @@ def __init__(
         if task_instance:
             self.dag_id = task_instance.dag_id
             self.task_id = task_instance.task_id
-            if execution_date := getattr(task_instance, "execution_date", None):
+            if execution_date := getattr(task_instance, "logical_date", None):
                 self.execution_date = execution_date
```

Comment:
```suggestion
            if logical_date := getattr(task_instance, "logical_date", None):
                self.execution_date = logical_date
```

I guess the column itself will be renamed in another PR?

Reply:
It’s probably better to just remove this. This field was only kept for compatibility and is no longer needed anywhere.

Reply:
I have changed it now. Now if `task_instance.logical_date` exists and is not None,it assigns this value to `self.logical_date`.